Remove styling from No Products filter results.

danielwrobert opened the pull request:
This is a small styling update that omits the `.woocommerce-info` styles on the filter results notification when no products match the filter.

See pdnLyh-1cy-p2 for additional context.

<!-- Reference any related issues or PRs here -->
Fixes #2026.

<!-- Briefly describe the issue or problem that this PR solves. -->

<!-- Explain your fix - how it addresses the problem, what else might be affected, any risks etc. -->

<!-- Don't forget to update the title with something descriptive. -->

### Screenshots
<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->
| Before | After |
| ------ | ----- |
|  ![CleanShot 2022-08-11 at 15 07 47](https://user-images.githubusercontent.com/481776/184220101-f53c93c9-017e-4542-ab33-ee454bb0d29f.png)    |  ![CleanShot 2022-08-11 at 15 05 09](https://user-images.githubusercontent.com/481776/184220090-ed30bdfe-c7c4-4240-b027-725fe2aff24a.png)   |


### How to test the changes in this Pull Request:
<!-- Add detailed instructions for how to test that this PR fixes the issue, and confirm that it doesn't break any other features :) -->

1. With the Storefront theme active, go to your site's Store page (displaying the WooCommerce Product Catalog) and add some filter blocks (e.g., Filter Products by Price, Filter Products by Attribute, etc.). You can add this directly on the page or in a widget area.
2. View the frontend and set your filters to where no products will match and be displayed.
3. Inspect the **No products** banner to see if it has a `woocommerce-no-products-found` class. If it does not, add that class via the inspector to see the changes in this PR take effect (the output for the banner comes from [this notice in WC Core](https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/templates/loop/no-products-found.php#L21) - see screenshot below).
4. Confirm the **No products** message appears as it does in the "After" screenshot above (with no blue banner styling).
5. Confirm no related JS or PHP errors.

![CleanShot 2022-08-17 at 16 02 48](https://user-images.githubusercontent.com/481776/185232649-67c80922-d0ab-40bc-b33e-ec2333c2557c.png)


<!-- Review the [flows & features doc](https://github.com/woocommerce/storefront/wiki/Testing-Storefront:-flows-and-features) and list any flows that might be impacted or improved -->


### Changelog

<!-- Add suggested changelog entry here. For example: -->

> Enhancement - Remove styling from No Products filter results. #2026

<!-- See [previous releases](https://github.com/woocommerce/storefront/releases) for more examples. -->

danielwrobert committed:
Remove styling from No Products filter results.

Omits the woocommerce-info styles on the filter results notification
when no products match the filter.

danielwrobert committed:
Fix linting errors.

While the linting errors are not related to this PR, they are failing
in the CI. This should fix the issues with the CSS and JS errors.

tjcafferkey commented:
If you go onto a product category page, and then apply some of the filters to activate the banner then the blue background is present here (`/product-category/clothing/accessories/?filter_color=yellow&query_type_color=or&max_price=13`)

I assume this is also the case on product tag pages too.

![Screenshot 2022-08-16 at 09 17 19](https://user-images.githubusercontent.com/8639742/184831967-b17a7a02-101d-418f-9286-f5cb0670accb.png)

danielwrobert committed:
Broaden selector to capture all store archives.

Instead of only targeting the WooCommerce store catalog, we also need to
target (omit) the info box on the archive pages as well, since the
archives all display lists of products under the woocommerce setions.

danielwrobert commented:
@tjcafferkey thanks for catching this! In 377f957, I've adjusted the selector to omit any of the archive pages that fall under the store (i.e., the Product post type).

danieldudzic commented:
Tested, and it works fine for me both on the shop and archive pages. Code also looks fine.

Aljullu commented:
I'm a bit concerned that this change might affect other notices besides the "No products were found..." one. I can't think of any examples, but there might be cases where WC or an extension add an info notice that needs to be visible in those pages but is not the one we want to remove the style from.

I wonder if we should opt for an alternative approach:

* Remove the `woocommerce-info` class from [this notice in WC core](https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/templates/loop/no-products-found.php#L21).
* Alternatively, add an extra class there so from Storefront we can target that specific notice and remove the styles only in that one.

Thoughts? :thinking:

danielwrobert commented:
@Aljullu you raise a good point, perhaps this has the potential to be a bit too broad.

I wouldn't recommend removing the class from that WC Core file because that would have an impact on all themes that rely on it, not just Storefront.

I do like the idea of adding a class there though (something like `woocommerce-no-products-found` or `woocommerce-no-products`). That way we can better ensure we're targeting the banner in the specific context that we expect.

I'll go ahead and create a PR and adjust this approach when that gets merged. Thanks!

_Note: converting back to Draft and setting status as `blocked` until the aforementioned adjustment is made in WC Core._

danielwrobert commented:
I created a small PR [here](https://github.com/woocommerce/woocommerce/pull/34362) to add a contextual class for the output.

I've adjusted the styling changes in 5078618 so that the changes will take effect when the WC Core PR is merged.

Since the changes are all in CSS and non-breaking, they shouldn't depend on one-another as far as merging goes.

I've also updated the test plan above.

danielwrobert committed:
Update selector to target new class in WC Core.

Added a new class to the "no products found" info output for us to
contextually target more reliably here.

See https://github.com/woocommerce/woocommerce/pull/34362.

tjcafferkey commented:
This is a better approach I think @danielwrobert, only thing I have noticed is that what I am seeing does not look like the screenshot you've posted in "After" because there is still padding around the notice which makes it not align with the title and look a bit strange. I think its due to the padding added in this commit https://github.com/woocommerce/storefront/pull/2027/commits/5078618b70c6c877316a729a2dc0a46822353607

![Screenshot 2022-08-22 at 11 22 01](https://user-images.githubusercontent.com/8639742/185899158-b421abfb-0d4b-4fe8-9bdd-52d0d2f449fa.png)

danielwrobert committed:
Update info message padding.

Zeroes out the left and right padding on the info banner so it aligns
with the heading. Also removes the unnecessary element portion of the
selector as to not be overly specific.

danielwrobert commented:
@tjcafferkey good eye, thanks for spotting that! I meant to zero out the side padding there. I've updated it in 93ad4f87.

danielwrobert commented:
Noting for posterity: @tjcafferkey as discussed in Slack the failing PHP test is unrelated to this PR and seems to be due to a bunch of [missing docblock comments](https://github.com/woocommerce/storefront/runs/7951989700?check_suite_focus=true).

There is an open Issue (#2004) to investigate this.
